70 definir intrfaz del módulo gestor de datos

IoachimusRoderici opened the pull request:
closes #61
closes #47 
closes #45 
closes #62

IoachimusRoderici committed:
renombrar excepción

IoachimusRoderici committed:
mover enum de días

IoachimusRoderici committed:
Crear dataclasses para los datos de aulas/edificios/clases

IoachimusRoderici committed:
crear gestor

IoachimusRoderici committed:
importar clases de excel

TeoInMaghine commented:
Hoy estuvimos investigando como vincular la base de datos con la interfaz gráfica, sacamos las siguientes conclusiones:
- El gestor de datos va a ser el que maneje la base de datos
- En el lado de la GUI, vamos a crear una clase que hereda de `QAbstractTableModel` que lee y escribe datos a través del gestor de datos

IoachimusRoderici committed:
Agregar carrera

IoachimusRoderici committed:
agregar horario a los edificios

IoachimusRoderici committed:
Juntar las definiciones en un archivo

IoachimusRoderici committed:
exportar

TeoInMaghine commented on src/asignacion_aulica/gestor_de_datos/clase.py:
Nevermind esto sería lo de este issue: #73

IoachimusRoderici committed:
Completar interfaz

TeoInMaghine commented on src/asignacion_aulica/gestor_de_datos/gestor.py:
Ah me olvidé en el anterior review que hice, creo que faltarían funciones `remove_*` para borrar entradas de las tablas.

IoachimusRoderici commented on src/asignacion_aulica/gestor_de_datos/entidades.py:
```diff
@@ -0,0 +1,77 @@
+
+@dataclass
+class Clase:
+    id: int # Identificador para distinguir entre distintas clases de la misma materia. No es para que lo vea el usuario.
```
El horario no sirve para identificar clases porque puede haber clases de la misma materia con el mismo horario (en días distintos o el mismo día pero de comisiones distintas).

Para identificar las clases de una materia sin usar un id, habría que usar la tupla `(día, comisión, horario)`. Me parece que eso es incómodo y que tiene más sentido asignarle un id a cada clase.

IoachimusRoderici committed:
Mover docente, auxiliar, y promocionable de materia a clase

IoachimusRoderici committed:
arreglar tipo de dato

IoachimusRoderici committed:
arreglar imports de día

IoachimusRoderici committed:
exportar cosas del gestor de datos

IoachimusRoderici committed:
Agregar métodos para borrar

IoachimusRoderici committed:
corregir caso de error en los docstrings

IoachimusRoderici committed:
Agregar validación de datos

IoachimusRoderici committed:
ignorar configuración de pyright

IoachimusRoderici committed:
borrado de datos jerárquico

IoachimusRoderici committed:
get_* devuelve las cosas ordenadas

IoachimusRoderici committed:
desambiguar sobreescritura y agregar callback de confirmación

IoachimusRoderici committed:
Cambiar Día por IntEnum con nombres en title case

Así el nombre de los elementos se puede usar como string y el valor se puede usar como índice
